src/main/java/it/eng/idsa/businesslogic/usagecontrol/service/impl/PlatoonUsageControlServiceImpl.java

```diff
@@ -65,12 +68,25 @@ public String uploadPolicy(String payloadContent) {
 		return communicationService.sendDataAsJson(ucUrl, payloadContent, CONTENT_TYPE);
 	}
 
+	@Override
+	public void rollbackPolicyUpload(String contractAgreement) {
```

Comment:
Why not pass here just UUID (as String) and do this serialize and get UUID outside this class; before making the call?

Reply:
We discussed to put it here in order to reduce clutter in the CH processor. Because this would otherwise be in that lamba function. If you want I can put it back?